src/components/TestToolMenu.js

```diff
@@ -43,12 +41,18 @@ const TestToolMenu = props => (
 
         {/* Option to switch from using the staging secure endpoint or the production secure endpoint.
         This enables QA and internal testers to take advantage of sandbox environments for 3rd party services like Plaid and Onfido. */}
-        <TestToolRow title="Use Staging Server">
-            <Switch
-                isOn={lodashGet(props, 'user.shouldUseStagingServer', _.contains([CONST.PLATFORM.WEB, CONST.PLATFORM.DESKTOP], getPlatform()))}
-                onToggle={() => User.setShouldUseStagingServer(!lodashGet(props, 'user.shouldUseStagingServer', true))}
-            />
-        </TestToolRow>
+        {
+            ApiUtils.canUseStagingToggle() && (
+                <TestToolRow title="Use Staging Server">
+                    <Switch
+                        isOn={lodashGet(props, 'user.shouldUseStagingServer', ApiUtils.isUsingStagingApi())}
+                        onToggle={() => User.setShouldUseStagingServer(
+                            !lodashGet(props, 'user.shouldUseStagingServer', ApiUtils.isUsingStagingApi()),
+                        )}
+                    />
+                </TestToolRow>
+            )
+        }
```

Comment:
Show the toggle only on STAGING, because the toggle doesn't work in DEV: https://github.com/Expensify/App/issues/15087#issuecomment-1431364109

Default the `isOn` value (when Onyx data is missing) to whether we currently use the staging api or not